src/store/cockpitApi.ts

```diff
@@ -92,6 +94,34 @@ export const cockpitApi = emptyCockpitApi.injectEndpoints({
           };
         },
       }),
+      getBlueprint: builder.query<GetBlueprintApiResponse, GetBlueprintApiArg>({
+        queryFn: async ({ id, version }) => {
+          try {
+            const blueprintsDir = await getBlueprintsPath();
+            const file = cockpit.file(path.join(blueprintsDir, id));
+
+            const contents = await file.read();
+            const parsed = toml.parse(contents);
+            file.close();
+
+            const blueprint = mapOnPremToHosted(parsed);
+
+            return {
+              data: {
+                ...blueprint,
+                id,
+                version,
+                last_modified_at: Date.now().toString(),
+                // TODO: get image requests. Will probably need
+                // to query cloudapi on composer socket.
+                image_requests: [],
```

Comment:
This is part of the createBlueprintRequest so with https://github.com/osbuild/image-builder-frontend/pull/2780 it should just be in the blueprint.